### PR TITLE
SC-8022 Custom Preview Size and some fixes

### PR DESCRIPTION
**Pull request by jhon-freitez-jemmic**
### Added
- Option to manually set the Preview Size and ignore the one returned by the server. This is useful for some ICAP Servers.
- Jenkinsfile.
- Jemmic's specific Gradle configuration.

### Removed
- Toolarium's common gradle.

### Improved
- Reading of the server's response preventing a "waiting" for data in the InputStream.
- Reading until a maximum number of bytes preventing some garbage bytes at the end of the Stream.
- `CHANGELOG.md`

### Fixed
- Options unit tests.

### CODE REVIEW

File: src/main/java/com/github/toolarium/icap/client/impl/ICAPClientImpl.java [hunk 1]
```diff
@@ -112,17 +112,17 @@ public ICAPRemoteServiceConfiguration options(final ICAPRequestInformation reque
             }
 
             boolean serverAllow204 = false;
-            if (icapHeaderInformation.containsHeader(ICAPConstants.HEADER_KEY_ALLOW) 
-                    && icapHeaderInformation.getHeaderValues(ICAPConstants.HEADER_KEY_ALLOW) != null 
+            if (icapHeaderInformation.containsHeader(ICAPConstants.HEADER_KEY_ALLOW)
+                    && icapHeaderInformation.getHeaderValues(ICAPConstants.HEADER_KEY_ALLOW) != null
                     && icapHeaderInformation.getHeaderValues(ICAPConstants.HEADER_KEY_ALLOW).size() > 0) {
                 serverAllow204 = Boolean.valueOf(icapHeaderInformation.getHeaderValues(ICAPConstants.HEADER_KEY_ALLOW).get(0).equalsIgnoreCase("204"));
```
Comment (christophe-kamphaus-jemmic) [inline src/main/java/com/github/toolarium/icap/client/impl/ICAPClientImpl.java:118]: Where is the information whether the server allows 204 overridden?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,20 +5,25 @@ All notable changes to this project will be documented in this file.
...
+
+### Removed
+- Toolarium's common gradle.
+
+### Improved
+- Reading of the server's response preventing a "waiting" for data in the InputStream.
+- Reading until a maximum number of bytes preventing some garbage bytes at the end of the Stream.
+- `CHANGELOG.md`
 
-## [ 1.0.2 ] - 2022-04-04
```
Comment (christophe-kamphaus-jemmic) [inline CHANGELOG.md:14]: Why did you remove the previous changelog entries?

File: src/main/java/com/github/toolarium/icap/client/impl/ICAPClientImpl.java (hunk 1 above)
Comment (jhon-freitez-jemmic) [reply src/main/java/com/github/toolarium/icap/client/impl/ICAPClientImpl.java:118]: This is the same to the original:
https://github.com/jemmic/icap-client/blob/10f2cd0b44b060058e557510b0c2820ab7e32546/src/main/java/com/github/toolarium/icap/client/impl/ICAPClientImpl.java#L389

Comment (christophe-kamphaus-jemmic) [reply src/main/java/com/github/toolarium/icap/client/impl/ICAPClientImpl.java:118]: Did the original already have the possibility to override that?

File: build.gradle [hunk 3]
```diff
@@ -1,23 +1,114 @@
...
+        }
+    }
+    dependencies {
+        classpath "com.palantir.gradle.gitversion:gradle-git-version:0.15.0"
+    }
+}
+
+repositories {
+    maven {
+        url "https://nexus.in.jemmic.com/repository/maven-public/"
```
Comment (christophe-kamphaus-jemmic) [inline build.gradle:70]: We cannot publish to `nexus.in.jemmic.com/repository/maven-public`.
We need to specify a separate repo for publishing.
